### DISCUSSION TITLE
Tailwind should move from apply to extend?

### DISCUSSION THREAD

**Discussion by sonic1981** [tailwindlabs/tailwindcss#7993] [Ideas]
I just came across [this issue](https://github.com/w3c/csswg-drafts/issues/1047) looking into an issue were having with `@apply`. It looks like the `@apply` rule is now deprecated in css according to w3c. Where as the similar, [`@extend`](https://github.com/w3c/csswg-drafts/issues/1047#issuecomment-828664467) is still active (although in draft).

This is also reflected in the [`postcss-apply` plugin docs in npm](https://www.npmjs.com/package/postcss-apply)

>The @apply rule and custom property sets most likely won't get any more support from browser vendors as the spec is yet considered deprecated and [alternative solutions](https://tabatkins.github.io/specs/css-shadow-parts) are being discussed.

Yet tailwind is still [recommending using `@apply`](https://tailwindcss.com/docs/reusing-styles#extracting-classes-with-apply).

Should this be changed?

**Comment by RobinMalfait** [comment]
Hey! Thank you for your question!
Much appreciated! 🙏

This should not be changed, the `@apply` directive is an API that _we_ expose and handle on our side before it ever hits the browser.

Similar to how `@tailwind utilities` is not a _native_ browser thing, it's also something we process.

In fact, you should be able to look at the produced css file in the browser and there should not be any `@tailwind`, `@apply` or anything like that. (unless you are referencing the source file instead of the dist file, but then nothing would work)

Hope this helps!

**Reply by sonic1981** [reply]
Well, you say that. But if I'm understanding correctly, tailwind doesn't process this. Postcss and specifically the postcss-import plugin processes this. They've stated that this is deprecated?
